automation_tools/__init__.py

```diff
@@ -1545,24 +1545,21 @@ def product_install(distribution, create_vm=False, certificate_url=None,
         # if we have ssh key to libvirt machine we can setup access to it
         if os.environ.get('LIBVIRT_KEY_URL') is not None:
             execute(setup_libvirt_key, host=host)
-        # setup_foreman_discovery for 6.1, 6.2 & upstream
+        # setup_foreman_discovery
         execute(
             setup_foreman_discovery,
             sat_version=satellite_version,
-            upstream=distribution.endswith('upstream'),
             host=host
         )
-        if not distribution.endswith('upstream'):
-            if satellite_version != '6.0':
-                execute(install_puppet_scap_client, host=host)
-            if satellite_version == '6.1':
-                execute(setup_oscap, host=host)
-            if satellite_version not in ('6.0', '6.1'):
-                execute(oscap_content, host=host)
+        if satellite_version != '6.0':
+            execute(install_puppet_scap_client, host=host)
+        if satellite_version == '6.1':
+            execute(setup_oscap, host=host)
+        if satellite_version not in ('6.0', '6.1'):
+            execute(oscap_content, host=host)
         # ostree plugin is for Sat6.2+ and nightly (rhel7 only)
```

Comment:
lines 1554 to 1559 either needs to be indented into one common ```if``` or yet better now all 3 ifs can merged with logic sat version != nightly
1. ```if satellite_version not in ('6.0', 'nightly'):``` (added or not nightly)
2. ```if satellite_version == '6.1':``` (stays same)
3. ```if satellite_version not in ('6.0', '6.1', 'nightly'):``` (added or not nightly)